src/libraries/System.Private.CoreLib/src/System/Guid.cs

```diff
@@ -49,24 +49,26 @@ namespace System
         }
 
         // Creates a new guid from a read-only span.
-        public Guid(ReadOnlySpan<byte> b)
+        public Guid(ReadOnlySpan<byte> b) : this(b, false) { }
```

Comment:
This is going to regress performance of this Guid ctor a lot on little endian systems.  For example, try to benchmark the following:

```
[Benchmark]
Guid GuidCtor()
{
   ReadOnlySpan<byte> buffer = new byte[] { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15 };
   return new Guid(buffer);
}
```

It is 2.5x slower compared to baseline.

Reply:
To balance the amount duplicate and performance, the implementation can look like this:
```
        // Creates a new guid from a read-only span.
        public Guid(ReadOnlySpan<byte> b)
        {
            if (b.Length != 16)
            {
                ThrowGuidArrayCtorArgumentException();
            }

            this = MemoryMarshal.Read<Guid>(b);

            if (!BitConverter.IsLittleEndian)
            {
                _a = BinaryPrimitives.ReverseEndianness(_a);
                _b = BinaryPrimitives.ReverseEndianness(_b);
                _c = BinaryPrimitives.ReverseEndianness(_c);
            }
        }

        public Guid(ReadOnlySpan<byte> b, bool bigEndian)
        {
            if (b.Length != 16)
            {
                ThrowGuidArrayCtorArgumentException();
            }

            this = MemoryMarshal.Read<Guid>(b);

            if (BitConverter.IsLittleEndian == bigEndian)
            {
                _a = BinaryPrimitives.ReverseEndianness(_a);
                _b = BinaryPrimitives.ReverseEndianness(_b);
                _c = BinaryPrimitives.ReverseEndianness(_c);
            }
        }

        [StackTraceHidden]
        private static void ThrowGuidArrayCtorArgumentException()
        {
            throw new ArgumentException(SR.Format(SR.Arg_GuidArrayCtor, "16"), "b");
        }
```
This will avoid regressing performance of the existing ctor on little endian systems, and it should improve performance of the existing ctor on big-endian systems. (Reading the bytes one at a time is slow.)

Reply:
I had suggested the inverse, implementing the new ctor in terms of the existing. The new one that takes a bool would be a tad more expensive, but it would avoid the duplication.

Reply:
Your suggestion looks good, too.